whois/table_rdap_domain.go

```diff
@@ -0,0 +1,66 @@
+package whois
+
+import (
+	"context"
+	"strings"
+
+	"github.com/openrdap/rdap"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+func tableRdapDomain(ctx context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "rdap_domain",
+		Description: "RDAP domain information including expiration, DNS servers and contact details.",
+		List: &plugin.ListConfig{
+			KeyColumns: plugin.SingleColumn("domain"),
+			Hydrate:    getRdapDomain,
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "domain", Type: proto.ColumnType_STRING, Transform: transform.FromQual("domain"), Description: "Domain name the RDAP information relates to."},
+
+			{Name: "domain_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("Handle").NullIfZero(), Description: "Unique identifier for the domain."},
+			{Name: "object_class_name", Type: proto.ColumnType_STRING, Description: "String 'domain' representing the object type in RDAP."},
+			{Name: "ldh_name", Type: proto.ColumnType_STRING, Description: "Textual representation of DNS names.", Transform: transform.FromField("LDHName")},
+			{Name: "port43", Type: proto.ColumnType_STRING, Description: "Hostname of Registry WHOIS server."},
```

Comment:
The description seems a bit strange based on the column name, do you have an example of the data we get back from the SDK for this field/column?

Reply:
@cbruno10, I queried this table  to retrieve the 'port43' value by using multiple domain names. However, in all cases, we received null or empty data.

I found the description mentioned in the [document](https://fred.nic.cz/documentation/html/RDAPReference/Domain.html), so I included it here.

```
> select domain, port43 from rdap_domain where domain = 'example.com'
+-------------+--------+
| domain      | port43 |
+-------------+--------+
| example.com |        |
+-------------+--------+

Time: 3.2s. Rows fetched: 1. Hydrate calls: 0.
> select domain, port43 from rdap_domain where domain = 'google.org'
+------------+--------+
| domain     | port43 |
+------------+--------+
| google.org |        |
+------------+--------+

Time: 4.5s. Rows fetched: 1. Hydrate calls: 0.
> select domain, port43 from rdap_domain where domain = 'microsoft.info'
+----------------+--------+
| domain         | port43 |
+----------------+--------+
| microsoft.info |        |
+----------------+--------+

Time: 2.2s. Rows fetched: 1. Hydrate calls: 0.
> select domain, port43 from rdap_domain where domain = 'whitehouse.gov'
+--------+--------+
| domain | port43 |
+--------+--------+
+--------+--------+

Time: 354ms.
> select domain, port43 from rdap_domain where domain = 'amazon.in'
+--------+--------+
| domain | port43 |
+--------+--------+
+--------+--------+

Time: 399ms.
> select domain, port43 from rdap_domain where domain = 'amazon.com'
+------------+--------+
| domain     | port43 |
+------------+--------+
| amazon.com |        |
+------------+--------+

Time: 1.2s. Rows fetched: 1. Hydrate calls: 0.
```